src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -317,6 +327,10 @@ bool ThetaStarPathPlanner::visitNeighbours(const Coordinate &current_coord,
     {
         for (int y_offset : {-1, 0, 1})
         {
+            if (i == 0 || i == num_grid_rows || j == 0 || j == num_grid_cols)
+            {
+                continue;
+            }
```

Comment:
I don't think this is necessary anymore

Reply:
At least in all the current test cases, we never run into this if branch, so it should be fine to remove it at least now. It is removed

Reply:
ah I think that's because updateVertex checks that the coordinate is valid